File: java/src/main/java/org/sparky/EnvironmentExternProvider.java
package org.sparky;

import org.sparky.model.ExternProvider;

/**
 * Created by lloyd on 02/03/2016.
 */
public class EnvironmentExternProvider implements ExternProvider{

    @Override
    public String getExtern(String key) {
        return System.getenv(key);
    }

    @Override
    public boolean hasExtern(String key) {
        return System.getenv(key) != null;
    }
}

File: net/Sparky/EnvironmentExternProvider.cs
using Sparky.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Sparky
{
    public class EnvironmentExternProvider : IExternProvider
    {
        public string GetExtern(string key)
        {
            return Environment.GetEnvironmentVariable(key);
        }

        public bool HasExtern(string key)
        {
            return Environment.GetEnvironmentVariable(key) != null;
        }
    }
}
